docs/source/basic_tutorials/execution.md

```diff
@@ -0,0 +1,128 @@
+<!--Copyright 2024 The HuggingFace Team. All rights reserved.
```

Comment:
I feel like this whole doc is just a repeat of https://moon-ci-docs.huggingface.co/docs/accelerate/pr_2456/en/concept_guides/deferring_execution no? I don't think it makes sense to have it here in the base tutorials.

Reply:
I think the "Deferring execution" guide shows examples of when to use `main_process_first` or `wait_for_everyone` such as downloading a dataset or saving/loading a state dict. This doc has a broader scope and introduces the different _execution scenarios_ you should consider (ie, do you want to execute on one process, on a specific process).

I think introducing these types of scenarios in the tutorials gives users a nice basic understanding of how to apply it to the more specific examples, such as the ones showcased in the Deferring execution doc. Let me know what you think! 🙂